src/OpenColorIO/fileformats/xmlutils/XMLReaderUtils.h

```diff
@@ -154,18 +156,31 @@ void ParseNumber(const char * str, size_t startPos, size_t endPos, T & value)
     }
 
     const char * startParse = str + startPos;
+    const char * adjustedParse = startParse;
 
     double val = 0.0f;
-    char * endParse = nullptr;
-
-    // The strtod expects a C string and str might not be null terminated.
-    // However since strtod will stop parsing when it encounters characters
-    // that it cannot convert to a number, in practice it does not need to
-    // be null terminated.
-    // C++11 version of strtod processes NAN & INF ASCII values.
-    val = strtod(startParse, &endParse);
+
+    // charconv's from_chars expects a C string and str might not be null terminated.
+    // However, since from_chars also takes a pointer to the last character,
+    // the string does not need to be null terminated explicitly.
+    // Do note, however, that plus signs and leading whitespace
+    // need to be skipped manually -- this
+    // was left implicit in the usage of strtod.
+    // TODO: allow hex numbers -- this will probably mean we need another from_chars library
+    const std::string parsedStr = StringUtils::LeftTrim({startParse, endPos - startPos});
+    if (startPos + parsedStr.size() != endPos) {
+        adjustedParse = str + endPos - parsedStr.size();
+    }
```

Comment:
The code should not use the StringUtils methods (which behind the scene mostly creates strings). Here is the code must only manipulate pointers.